includes/class-wc-payments-token-service.php

```diff
@@ -103,6 +126,58 @@ public function woocommerce_get_customer_payment_tokens( $tokens, $user_id, $gat
 
 		$customer_id = $this->customer_service->get_customer_id_by_user_id( $user_id );
 
+		$tokens = $this->migrate_customerless_tokens( $tokens, $user_id );
+		$tokens = $this->import_customer_tokens( $tokens, $customer_id, $user_id );
+		$tokens = $this->remove_unavailable_tokens( $tokens, $customer_id );
+
+		return $tokens;
+	}
+
+	/**
+	 * Delete token from Stripe.
+	 *
+	 * @param string           $token_id Token ID.
+	 * @param WC_Payment_Token $token    Token object.
+	 */
+	public function woocommerce_payment_token_deleted( $token_id, $token ) {
+		if ( WC_Payment_Gateway_WCPay::GATEWAY_ID === $token->get_gateway_id() ) {
+			try {
+				$this->payments_api_client->detach_payment_method( $token->get_token() );
+				// Clear cached payment methods.
+				$this->customer_service->clear_cached_payment_methods_for_user( $token->get_user_id() );
+			} catch ( Exception $e ) {
+				Logger::log( 'Error detaching payment method:' . $e->getMessage() );
+			}
+		}
+	}
+
+	/**
+	 * Set as default in Stripe.
+	 *
+	 * @param string           $token_id Token ID.
+	 * @param WC_Payment_Token $token    Token object.
+	 */
+	public function woocommerce_payment_token_set_default( $token_id, $token ) {
+		if ( WC_Payment_Gateway_WCPay::GATEWAY_ID === $token->get_gateway_id() ) {
+			$customer_id = $this->customer_service->get_customer_id_by_user_id( $token->get_user_id() );
+			if ( $customer_id ) {
+				$this->customer_service->set_default_payment_method_for_customer( $customer_id, $token->get_token() );
+				// Clear cached payment methods.
+				$this->customer_service->clear_cached_payment_methods_for_user( $token->get_user_id() );
+			}
+		}
+	}
+
+	/**
+	 * Imports customer payment methods from the API and adds them to WooCommerce.
+	 *
+	 * @param array  $tokens      Token list.
+	 * @param string $customer_id Customer ID.
+	 * @param int    $user_id     User ID.
+	 *
+	 * @return array Token list with imported tokens.
+	 */
+	private function import_customer_tokens( $tokens, $customer_id, $user_id ) {
```

Comment:
Great catch, I've added a catch block logging the error in https://github.com/Automattic/woocommerce-payments/pull/973/commits/34f9c165d49ae048f637d86f0f32b9d3eb51f17c to avoid breaking the checkout and payment methods pages.